### PR TITLE
Add sbsigntool package.

### PR DESCRIPTION
**Pull request by L38s**
# Purpose of pull request

sbsigntool 

This pr adds sbsigntool package based on meta-luv's recipes-devtools/sbsigntool/sbsigntool_git.bb

# Test
## How to test

I used this site as a reference.
https://wiki.archlinux.jp/index.php/%E3%82%BB%E3%82%AD%E3%83%A5%E3%82%A2%E3%83%96%E3%83%BC%E3%83%88

Build in qemu64 environment.
Add following line to conf/local.conf
```
IMAGE_INSTALL_append = " sbsigntool openssh openssl-bin"
IMAGE_ROOTFS_EXTRA_SPACE_append = " + 1000000"
```
Do bitbake, and when you are done, start QEMU with the following command
```
MACHINE=qemuarm64 runqemu nographic qemuparams="-m 8192 -smp 4"
```
Once started, transfer the files from the host machine.
```
$ scp ${BUILDDIR}/tmp-glibc/deploy/images/qemuarm64/Image--git0+ad19e133ae-r0-qemuarm64-<datetime>.bin root@192.168.7.2:~/Image
```
`${BUILDDIR}`: BUILDDIR is yocto's build directory
`<datetime>`: The datetime will change depending on the build.
~~~
# openssl req -newkey rsa:2048 -nodes -keyout MOK.key -new -x509 -sha256 -days 3650 -subj "/CN=my Machine Owner Key/" -out MOK.crt -config /etc/ssl/openssl.cnf
# openssl x509 -outform DER -in MOK.crt -out MOK.cer
# sbsign --key MOK.key --cert MOK.crt --output Image.signed Image
# sbverify --list Image.signed
~~~

## Test result

```
root@qemuarm64:~# openssl req -newkey rsa:2048 -nodes -keyout MOK.key -new -x509 -sha256 -days 3650 -subj "/CN=my Machine Owner Key/" -out MOK.crt -config /etc/ssl/openssl.cnf
root@qemuarm64:~# openssl x509 -outform DER -in MOK.crt -out MOK.cer
root@qemuarm64:~# sbsign --key MOK.key --cert MOK.crt --output Image.signed Image
Signing Unsigned original image
root@qemuarm64:~# sbverify --list Image.signed
signature 1
image signature issuers:
 - /CN=my Machine Owner Key
image signature certificates:
 - subject: /CN=my Machine Owner Key
   issuer:  /CN=my Machine Owner Key
```

### CODE REVIEW

File: recipes-debian/sbsigntool/sbsigntool/0001-configure-Fixup-build-dependencies-for-cross-compili.patch [hunk 1]
```diff
@@ -0,0 +1,54 @@
+From c3533b8da1e1425801d2fc0bcd231e13d593f16b Mon Sep 17 00:00:00 2001
```
Comment (masami256) [inline recipes-debian/sbsigntool/sbsigntool/0001-configure-Fixup-build-dependencies-for-cross-compili.patch:1]: Where did this patch come from ? I couldn't find exact this patch in [meta-luv](http://git.yoctoproject.org/cgit/cgit.cgi/meta-luv/tree/recipes-devtools/sbsigntool) nor [luv-yocto/meta-luv](https://github.com/intel/luv-yocto/tree/warrior/meta-luv/recipes-devtools/sbsigntool/sbsigntool).

File: recipes-debian/sbsigntool/sbsigntool_debian.bb [hunk 2]
```diff
@@ -0,0 +1,52 @@
...
+    file://disable-man-page-creation.patch \
+    file://0001-configure-Fixup-build-dependencies-for-cross-compili.patch \
+    "
+
+BBCLASSEXTEND = "native"
+
+DEPENDS = "binutils openssl gnu-efi util-linux"
+
+do_configure_class-native() {
+    if [ ! -e lib/ccan.git/Makefile ]; then
```
Comment (masami256) [inline recipes-debian/sbsigntool/sbsigntool_debian.bb:24]: According to the debian source package,  debian source doesn't contain .git directory. so this code seems doesn't need it.

```
masami@ubuntu1804:~/tmp/sbsigntool-0.9.2/lib$ pwd
/home/masami/tmp/sbsigntool-0.9.2/lib
masami@ubuntu1804:~/tmp/sbsigntool-0.9.2/lib$ find . -name .git
masami@ubuntu1804:~/tmp/sbsigntool-0.9.2/lib$ 
```

File: recipes-debian/sbsigntool/sbsigntool_debian.bb [hunk 3]
```diff
@@ -0,0 +1,52 @@
...
+                talloc read_write_all build_assert array_size
+    fi
+
+    export CC="${OLD_CC}"
+    export CFLAGS="-I${STAGING_INCDIR}/efi -I${STAGING_INCDIR}/efi/x86_64"
+    ./autogen.sh --noconfigure
+    oe_runconf
+}
+
+patch() {
```
Comment (masami256) [inline recipes-debian/sbsigntool/sbsigntool_debian.bb:50]: Would you change function name of the patch()? Yocto has patch() so this name makes bit confused.

File: recipes-debian/sbsigntool/sbsigntool/0001-configure-Fixup-build-dependencies-for-cross-compili.patch (hunk 1 above)
Comment (L38s) [reply recipes-debian/sbsigntool/sbsigntool/0001-configure-Fixup-build-dependencies-for-cross-compili.patch:1]: https://lists.01.org/hyperkitty/list/luv@lists.01.org/thread/RY2WY2COCNZ6QD2HCZI2XBQO5VE7U6QK/
I applied the patch based on the information above and the patch applied by meta-intel's sbsigntool, and it solved the problem.

Comment (masami256) [reply recipes-debian/sbsigntool/sbsigntool/0001-configure-Fixup-build-dependencies-for-cross-compili.patch:1]: @L38s I got it. This recipe is also based on the patch [[PATCH] sbsigntool: Use native sysroot to build], right? If so, please describe it so by comment in the recipe.

File: recipes-debian/sbsigntool/sbsigntool_debian.bb (hunk 3 above)
Comment (masami256) [reply recipes-debian/sbsigntool/sbsigntool_debian.bb:50]: It's okay. this recipe is based on [thie one](https://lists.01.org/hyperkitty/list/luv@lists.01.org/thread/RY2WY2COCNZ6QD2HCZI2XBQO5VE7U6QK/)

**Comment by L38s** [comment]
Sorry, I failed to manage the commit, so I'll close it once and PR it again.

**Pull request closed without merging**